wallet/src/error.rs

```diff
@@ -1,14 +1,16 @@
 // Copyright 2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use std::fmt::{Debug, Display};
+use std::fmt::Debug;
 
 use iota_client::block::payload::transaction::TransactionId;
-use serde::{ser::Serializer, Serialize};
+use serde::{
+    ser::{SerializeMap, Serializer},
+    Serialize,
+};
 
 /// The wallet error type.
-#[derive(Debug, thiserror::Error, Serialize)]
-#[serde(tag = "type", content = "error", rename_all = "camelCase")]
```

Comment:
Wondering, what was this doing if not what we want?

Reply:
The type/error part I mean

Reply:
The `error` field didn't use the Display string, but just the inner of the variant
So if you have an error with these variants
```
    #[error("first letter must be lowercase")]
    MissingIndex,
    #[error("invalid index {index}")]
    InvalidIndex { index: u8 },
    #[error("invalid number {0}")]
    InvalidNumber(u8),
```
and serialize it, the result looks like
```
{"type":"invalidIndex","error":{"index":5}}
{"type":"missingIndex"}
{"type":"invalidNumber","error":1}
```
with the change it looks like
```
{"type":"invalidIndex","error":"invalid index 5"}
{"type":"missingIndex","error":"first letter must be lowercase"}
{"type":"invalidNumber","error":"invalid number 1"}
```